sys-apps/pciutils/pciutils-3.6.0.recipe

```diff
@@ -0,0 +1,140 @@
+the system
+- setpci - allows reading from and writing to PCI device configuration \
+registers. For example, you can adjust the latency timers with it."
+HOMEPAGE="https://mj.ucw.cz/sw/pciutils/"
+COPYRIGHT="1997-2018 Martin Mares"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="ftp://atrey.karlin.mff.cuni.cz/pub/linux/pci/pciutils-$portVersion.tar.gz"
+CHECKSUM_SHA256="10e9cdeb8d147051dc6aa7f3aa4ae086ba516ce8f81357d6b2d0da9c5923531b"
+PATCHES="pciutils-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+# libpci.so can only be built (for this package) if gcc >= 4.
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+	LIBPCI_SHARED=yes
+else
+	LIBPCI_SHARED=no
+fi
+
+PROVIDES="
+	pciutils$secondaryArchSuffix = $portVersion
+	cmd:lspci$secondaryArchSuffix = $portVersion
+	cmd:setpci$secondaryArchSuffix = $portVersion
+	"
+if [ "$LIBPCI_SHARED" = yes ]; then
+	PROVIDES="$PROVIDES
+		lib:libpci$secondaryArchSuffix = $portVersion compat >= 3
+		"
+fi
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	pciutils_data == $portVersion
+	"
+
+if [ -z "$secondaryArchSuffix" ]; then
+	SUMMARY_data="Utilities for inspecting and manipulating PCI devices \
+(data)"
+	ARCHITECTURES_data="any"
+	PROVIDES_data="
+		pciutils_data = $portVersion
+		"
+	REQUIRES_data=""
+fi
+
+PROVIDES_devel="
+	pciutils${secondaryArchSuffix}_devel = $portVersion
+	devel:libpci$secondaryArchSuffix = $portVersion compat >= 3
+	"
+REQUIRES_devel="
+	pciutils$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:sed
+	"
+
+
+BUILD()
+{
+	make \
+		ZLIB=no \
+		SHARED=$LIBPCI_SHARED \
+		PREFIX=$prefix \
+		SHAREDIR=$dataDir \
+		IDSDIR=$dataDir/pciutils \
+		SBINDIR=$binDir \
+		MANDIR=$manDir \
+		INCDIR=$includeDir \
+		LIBDIR=$libDir
+}
+
+INSTALL()
+{
+	make \
+		ZLIB=no \
+		SHARED=$LIBPCI_SHARED \
+		PREFIX=$prefix \
+		SHAREDIR=$dataDir \
+		IDSDIR=$dataDir/pciutils \
+		SBINDIR=$binDir \
+		MANDIR=$manDir \
+		INCDIR=$includeDir \
+		LIBDIR=$libDir \
+		install install-lib
+
+	# update-pciids is useless because pci.ids is read-only.
+	rm -f $binDir/update-pciids
+	rm -f $manDir/man*/update-pciids.*
+
+	prepareInstalledDevelLib libpci
+	fixPkgconfig
+
+	if [ -z "$secondaryArchSuffix" ]; then
+		maybe_manDir_man7=$manDir/man7
+		packageEntries data \
+			$dataDir
+		sed -i -e "s,$dataDir/,/system/data/,g" \
+			$manDir/man8/lspci.8 \
+			$manDir/man8/setpci.8
+		local include_pci=/system/$relativeOldIncludeDir/pci
+		sed -i -e "s,/usr/include/linux/,$include_pci/,g" \
+			$manDir/man8/lspci.8
+		sed -i -e "s,/usr/include/pci/,$include_pci/,g" \
+			$manDir/man8/setpci.8
+	else
+		maybe_manDir_man7=
+		rm -rf $dataDir $documentationDir
+	fi
+
+	# $dataDir/pciutils/pci.ids is not in pciutils but in pciutils_data.
+	local data_dir=/packages/pciutils_data${dataDir#/packages/pciutils$secondaryArchSuffix}
```

Comment:
```diff
-	local data_dir=/packages/pciutils_data${dataDir#/packages/pciutils$secondaryArchSuffix}
+	local data_dir=$(getPackagePrefix pciutils_data-$portVersion)/${relativeDataDir}
```

This should looks cleaner :)

Reply:
Hmm, I would be OK with using `getPackagePrefix`, but not until someone has managed to fix haikuporter so that it won't produce the warnings we get in Travis-CI.
We already have `getPackagePrefix` in a few other recipes, and everytime the Travis-CI script is called for any recipe, we see a lot of noise:
<Details>

```
packageEntries: error: "fortran" doesn't seem to be
a valid package suffix.
packageEntries: error: "fortran" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
packageEntries: error: "syslibs" doesn't seem to be
a valid package suffix.
```

Reply:
I think it's only a problem when you put it on the topmost level. Travis certainly won't care about stuff in functions.